### ISSUE TITLE
mailer - endpoint invocation stays blocked when having mailer misconfigured

### ISSUE THREAD

**Issue by rsvoboda** [quarkusio/quarkus#27919]
### Describe the bug

mailer - endpoint invocation stays blocked when having mailer misconfigured

when I used config based on https://quarkus.io/guides/mailer#configuring-the-mailer and invoked `curl localhost:8080/mail/` I got warning and error in log
```
2022-09-14 09:54:06,624 WARN [io.ver.ext.mai.imp.SMTPAuthentication] (vert.x-eventloop-thread-0) io.vertx.ext.mail.SMTPException: AUTH LOGIN failed: 535 Authentication failed: Bad username / password
2022-09-14 09:54:06,634 ERROR [io.ver.ext.mai.imp.SMTPConnection] (vert.x-eventloop-thread-0) SSLEngine closed already
```

But curl command didn't finish, still waiting for response.
And I had to interrupt it using `Ctrl+C`

Endpoint invocation shouldn't be blocked, looks like Quarkus or vertx-mail-client issue.

https://quarkus.io/guides/mailer#configuring-the-mailer based config
```properties
quarkus.mailer.mock=false
quarkus.mailer.from=test@quarkus.io
quarkus.mailer.host=smtp.sendgrid.net
quarkus.mailer.port=465
quarkus.mailer.ssl=true
quarkus.mailer.username=foo
quarkus.mailer.password=bar
```

When using miss-configured gmail setting I received `{"details":"Error id ea7bfd08-69cf-4772-9b3b-f640a2de938f-1, java.util.concurrent.CompletionException: io.vertx.ext.mail.SMTPException: AUTH PLAIN failed: 535-5.7.8 Username and Password not accepted. Learn more at","stack":"java.util.concurrent.CompletionException: io.vertx.ext.mail.SMTPException: AUTH PLAIN failed: 535-5.7.8 Username and Password not accepted. Learn more at\n535 5.7.8  https://support.google.com/mail/?p=BadCredentials h20-200 .... `

So maybe sendgrid doesn't provide any text response like gmail and thus quarkus app still keeps the connection open / curl command doesn't finish. but logs are indicating that authentication wasn't successful and thus the connection for curl should be closed with appropriate response code / response text.

### Expected behavior

endpoint invocation doesn't stay blocked

### Actual behavior

endpoint invocation stays blocked

### How to Reproduce?

 - use mailer-quickstart
 - adjust application.properties
   - config described above
 - run the app - e.g. `mvn clean quarkus:dev`
 - run `curl localhost:8080/mail`

### Output of `uname -a` or `ver`

macOS Monterey

### Output of `java -version`

Java 17

### Quarkus version or git rev

Quarkus main

**Comment by gsmet** [comment]
Probably a good idea to get a stacktrace of the stuck thread.

**Comment by rsvoboda** [comment]
Full thread dump by jstack command - [jstack.txt](https://github.com/quarkusio/quarkus/files/9565419/jstack.txt)

**Comment by cescoffier** [comment]
So... there is a bug and a workaround.
The is a bug in the Vert.x Mailer which emits auth command asynchronously when there is a chain, but seems to forget to report in some cases. I will open an issue on the Vert.x mailer (@vietj for your info).

The workaround  consists of not using auth chain, and just adding:
```
quarkus.mailer.auth-methods=LOGIN
```

**Comment by cescoffier** [comment]
Issue opened: https://github.com/vert-x3/vertx-mail-client/issues/186

**Comment by cescoffier** [comment]
Fixed proposed: https://github.com/vert-x3/vertx-mail-client/pull/187.

**Comment by cescoffier** [comment]
Closed by #28323
